src/main/java/seedu/address/model/Database.java

```diff
@@ -228,6 +236,24 @@ public boolean equals(Object other) {
                 && patients.equals(otherDatabase.patients);
     }
 
+    /**
+     * Checks if the appointment is valid to be inserted into the appointment list
+     * @throws CommandException if {@code predicate} is null.
+     */
+    void validateAddAppointment(Appointment reference, Appointment toAdd) throws CommandException {
```

Comment:
overload this method with another one that takes in 1 argument only. The new method will call this method and pass null as reference. A user shouldnt need to know that he is supposed to pass null to this method